linkml/generators/docgen.py

```diff
@@ -1008,6 +1009,12 @@ def example_object_blobs(self, class_name: str) -> List[Tuple[str, str]]:
     default=True,
     help="Render class table on index page in a hierarchically indented view",
 )
+@click.option(
+    "--render-imports/--no-render-imports",
+    default=False,
```

Comment:
I have selected the default exactly to keep current interfaces unchanged. If `render_imports` is left with its default, then `merge_imports` is decisive: 

1. If you `--mergeimports`, then imports are seen as part of the schema (not anymore as imports) and will be rendered anyway, so `render_imports` does not play any role.
2. If you `--no-mergeimports`, since `render_imports` is per default to False, then you will not see the elements of the imported schemas. And that is exactly how it behaves as of now, since `index.md.jinja2` relies on `all_class_objects`, which [calls `all_classes`](https://github.com/linkml/linkml/blob/c4379e5a121732b6adda21284bb34dde590743b1/linkml/generators/docgen.py#L762) with what I have called `render_imports` == `merge_imports` (and therefore it is False).

Reply:
I will extend the newly added [`test_scripts/test_docgen.py`](https://github.com/linkml/linkml/pull/2365/files#diff-ac394de9c5462b93cfdbb052a5248aa93cf78aa708a5c4a1cdf63808f38570b6) so that it tests the visibility of imported elements depending on the value of `merge_imports`. I will put it in a commit before my changes and test it. That test unchanged should then further succeed after the addition of `render_imports`. That way we will have a confirmation for my logic 🙂

Reply:
Done.

@sujaypatil96 commit https://github.com/linkml/linkml/pull/2365/commits/cc65d1deed266fc2ba819119ec678dc0b5df117d provides a test that helps addressing this concern. It confirms that the imported class [`agent`](https://github.com/linkml/linkml/blob/c4379e5a121732b6adda21284bb34dde590743b1/tests/test_generators/input/core.yaml#L48) is [only reachable when imports are merged](https://github.com/linkml/linkml/pull/2365/commits/cc65d1deed266fc2ba819119ec678dc0b5df117d#diff-ac394de9c5462b93cfdbb052a5248aa93cf78aa708a5c4a1cdf63808f38570b6R30-R42). 

You can test it locally like this: `git checkout c4379e5a && pytest tests/tests_scripts/tests_gen_doc.py`

That very same test is still passing after my patch thanks to the fact that `render_imports` default to `False`.

Test [`test_render_imports`](https://github.com/linkml/linkml/pull/2365/files#diff-ac394de9c5462b93cfdbb052a5248aa93cf78aa708a5c4a1cdf63808f38570b6R93-R115) demonstrates that setting `render_imports` defaulting to `True` results in imported elements becoming rendered, what breaks current behavior.

I hope I could convince hereby of the correctness of having `render_imports` defaulting to `False` for backwards compatibility reasons.

Reply:
@sujaypatil96 it looks as if it is the only open point before merging this PR. Please let me know:

1. If I can resolve this conversation (you can also do it yourself, since you opened it).
2. Who takes care of merging it. I have the needed rights, but I am still unsure about the process 🫣

Reply:
from dev call:
- we'll try to merge ASAP during the week, using Friday dev call to sweep up PR merges.  If you have an urgent need to merge, just ping slack channel and we'll find someone to review and merge.
- @sujaypatil96 is ready to merge this one 👍

Reply:
> we'll try to merge ASAP during the week, using Friday dev call to sweep up PR merges. If you have an urgent need to merge, just ping slack channel and we'll find someone to review and merge.

Is this the expected workflow? Is it documented somewhere? That would be helpful.

Reply:
definitely - I took an action item to get it into our Contributing doc (which likely needs a bump anyway).